src/extension.ts

```diff
@@ -141,27 +133,52 @@ function getPRsPerSHAs() {
           "No search results. Try selecting a bigger piece of code or another file."
         );
       } else {
-        issuesBySHAs.forEach((issue: { url: any }) => {
-          const issueUrl = issue.url;
+        let issuesWithTitlesAndGroupedComments: { user: any; title: string; comments: any[]; created_at: any; }[] = [];
 
-          octokit
-            .request(`GET ${issueUrl}/comments`)
-            .then((octorespComments: any) => {
-              // this paints the panel
-              watermelonPanel.currentPanel?.doRefactor({
-                command: "prs",
-                data: octorespComments.data,
+        issuesBySHAs.forEach(async (issue: { url: any }) => {
+          const issueUrl = issue.url;
+          let prTitlesPushed: string[] = [];
+
+          await octokit.request(`GET ${issueUrl}/comments`).then(async (octoresp: { data: { issue_url: any; body: string; user: any; title: string; comments: any[]; created_at: any; }[]; }) => {
+            // this paints the panel
+            octoresp.data.forEach(async (issue: { issue_url: any, body: string; user: any; title: string; comments: any[]; created_at: any; }) => {
+              const issueUrl = issue.issue_url;
+  
+              await octokit.request(`GET ${issueUrl}`).then((octoresp2: { data: { title: string; }; }) => {
+                let prTitle ="";
+                prTitle = octoresp2.data.title;
+                issue.title = prTitle;
+  
+                if (prTitlesPushed.includes(prTitle)) {
+                  for (let i=0; i<issuesWithTitlesAndGroupedComments.length; i++) {
```

Comment:
Is this a NxN for?
Why do we do it so much?
Why not use indices or the comment id?